modules/oe_content_tender/tests/src/Kernel/TenderNodeWrapperTest.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_content_tender\Kernel;
+
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\entity_test\Entity\EntityTest;
+use Drupal\node\Entity\Node;
+use Drupal\node\Entity\NodeType;
+use Drupal\oe_content\EntityWrapperInterface;
+use Drupal\oe_content_tender\TenderNodeWrapper;
+use Drupal\oe_content_tender\TenderNodeWrapperInterface;
+
+/**
+ * Tests tender wrapper class.
+ *
+ * @coversDefaultClass \Drupal\oe_content_tender\TenderNodeWrapper
+ * @group oe_content_tender
+ */
+class TenderNodeWrapperTest extends TenderKernelTestBase {
+
+  /**
+   * Test exception thrown if an unsupported entity type is passed.
+   */
+  public function testUnsupportedEntityType(): void {
+    $this->expectException(\InvalidArgumentException::class);
+    $this->expectExceptionMessage("The current wrapper only accepts 'node' entities.");
+    $entity = EntityTest::create();
+    TenderNodeWrapper::getInstance($entity);
+  }
+
+  /**
+   * Test exception thrown if an unsupported entity bundle is passed.
+   */
+  public function testUnsupportedEntityBundle(): void {
+    $this->expectException(\InvalidArgumentException::class);
+    $this->expectExceptionMessage("The current wrapper only accepts 'node' entities of type 'oe_tender'.");
+    $node_type = NodeType::create([
+      'type' => 'test',
+      'name' => 'Test bundle',
+    ]);
+    $node_type->save();
+    $entity = Node::create([
+      'type' => 'test',
+    ]);
```

Comment:
Also, this should be saved. Currently, is only created.

Reply:
This is actually ok, here we are testing unsupported bundle.